src/box/wal_retention_delay.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "trivia/config.h"
+
+#if defined(ENABLE_WAL_RETENTION_DELAY)
+# include "wal_retention_delay_impl.h"
+#else /* !defined(ENABLE_WAL_RETENTION_DELAY) */
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+#include "vclock/vclock.h"
+#include "xlog.h"
+
+/** Initialize wal_retention_delay. */
+static inline void
+wal_retention_delay_create(struct xdir *wal_dir, struct xlog *current_wal,
+			   struct cpipe *tx_prio_pipe)
+{
+	(void)wal_dir;
+	(void)current_wal;
+	(void)tx_prio_pipe;
+}
+
+/** Cleanup wal_retention_delay module. */
+static inline void
+wal_retention_delay_destroy(void)
```

Comment:
Nit: Subsystem constructor and destructor should be named `X_init` and `X_free`, e.g. `wal_init` and `wal_free`.